component/src/main/java/org/wso2/extension/siddhi/execution/env/GetOriginIPFromXForwardedFunction.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.execution.env;
+
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.ReturnAttribute;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.executor.function.FunctionExecutor;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class provides a function to get immediate public IP of the origin from X-Forwarded-For header.
+ */
+@Extension(
+        name = "getOriginIPFromXForwarded",
+        namespace = "env",
+        description = "This function returns the public origin IP from the given X-Forwarded header",
+        returnAttributes = @ReturnAttribute(
+                description = "public IP related to the origin which is retrieved using the given X-Forwarded header",
+                type = {DataType.STRING}),
+        parameters = {
+                @Parameter(name = "xforwardedheader",
+                        description = "X-Forwarded-For header of the request",
+                        type = {DataType.STRING}
+                )
+        },
+        examples = {
+                @Example(
+                        syntax = "define stream inputStream (xForwardedHeader string);\n" +
+                                "from inputStream env:getOriginIPFromXForwarded(xForwardedHeader) as originIP \n" +
+                                "insert into outputStream;",
```

Comment:
"from InputStream \n" +
"select env:getOriginIPFromXForwarded(xForwardedHeader) as originIP \n" +
"insert into OutputStream;",

Note : Streams should start in Capital you have missed select